dart/common/LocalResource.cpp

```diff
@@ -0,0 +1,139 @@
+#include <cstring>
+#include <iostream>
+#include "dart/common/Console.h"
+#include "LocalResource.h"
+
+namespace dart {
+namespace common {
+
+LocalResource::LocalResource(const std::string& _path)
+  : mFile(std::fopen(_path.c_str(), "rb"))
+{
+  if(!mFile)
+  {
+    dtwarn << "[LocalResource::constructor] Failed opening file '" << _path
+          << "' for reading: " << std::strerror(errno) << "\n";
+  }
+}
+
+LocalResource::~LocalResource()
+{
+  if (!mFile)
+    return;
+
+  if (std::fclose(mFile) == EOF)
+  {
+    dtwarn << "[LocalResource::destructor] Failed closing file.\n";
+  }
+}
+
+bool LocalResource::isGood() const
+{
+  return !!mFile;
+}
+
+size_t LocalResource::getSize()
+{
+  if(!mFile)
+    return 0;
+
+  const long offset = std::ftell(mFile);
+  if(offset == -1L)
+  {
+    dtwarn << "[LocalResource::getSize] Unable to compute file size: Failed"
+             " getting current offset: " << std::strerror(errno) << "\n";
+    return 0;
+  }
+
+  // The SEEK_END option is not required by the C standard. However, it is
+  // required by POSIX.
+  // TODO: Does this work on Windows?
+  if(std::fseek(mFile, 0, SEEK_END) || std::ferror(mFile))
+  {
+    dtwarn << "[LocalResource::getSize] Unable to compute file size: Failed"
+             " seeking to the end of the file.\n";
```

Comment:
The [documentation I found](http://en.cppreference.com/w/cpp/io/c/fseek) says only that:

> If a read or write error occurs, the error indicator for the stream (`std::ferror`) is set and the file position is unaffected.

I don't think there is any guarantee that the value returned by `ferror` has the same meaning as `errno`. 

Again, the documentation [that you found](http://linux.die.net/man/3/fseek) says that `fseek` sets `errno`. I'm not sure what the deal is.